pkg/kubectl/resource_printer.go

```diff
@@ -84,15 +86,15 @@ func GetPrinter(format, formatArgument string) (ResourcePrinter, bool, error) {
 // ResourcePrinter is an interface that knows how to print runtime objects.
 type ResourcePrinter interface {
 	// Print receives a runtime object, formats it and prints it to a writer.
-	PrintObj(runtime.Object, io.Writer) error
+	PrintObj(runtime.Object, *watch.EventType, io.Writer) error
 }
 
 // ResourcePrinterFunc is a function that can print objects
-type ResourcePrinterFunc func(runtime.Object, io.Writer) error
+type ResourcePrinterFunc func(runtime.Object, *watch.EventType, io.Writer) error
```

Comment:
Alternatively, if composition isn't going to work (ResourcePrinter that invokes a ColumnResourcePriner), then there would need to be some way to pass in some info opaquely to each row/object you print that is "add this column for each item".  I'm not sure that's cleaner, because ResourcePrinter expands to take opaque context that is different for each implementer of the interface - if the implementer doesn't understand the context there may be no way to present the info.